lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -512,21 +548,47 @@ class ModuleEmitter : public EmitterBase {
 
   StringRef getName(Value value) { return getName(ValueOrOp(value)); }
   StringRef getName(ValueOrOp valueOrOp) {
-    auto *entry = nameTable[valueOrOp];
-    assert(entry && "value expected a name but doesn't have one");
-    return entry->getKey();
+    auto entry = nameTable.find(valueOrOp);
+    assert(entry != nameTable.end() &&
+           "value expected a name but doesn't have one");
+    return entry->getSecond();
+  }
+
+  /// Given a module name \p moduleName, return the updated name if it has been
+  /// previously renamed in the table moduleNameTable, else call
+  /// resolveKeywordConflict, to get a new name in case of conflict with
+  /// keywords.
+  StringRef getModuleName(StringAttr moduleName) {
+    auto entryIter = moduleNameTable.find(moduleName);
+    StringRef updatedName;
+    if (entryIter == moduleNameTable.end()) {
+      updatedName = resolveKeywordConflict(
+          moduleName.getValue(), usedModuleNames, nextGeneratedNameID);
+      moduleNameTable[moduleName] = updatedName;
+    } else
+      updatedName = entryIter->getSecond();
+    return updatedName;
```

Comment:
I'd personally invert this and merge the return into the if condition, e.g.:

if (entryIter != moduleNameTable.end())
  return entryIter->getSecond();
other stuff


but YMMV; anything is fine here.

Reply:
Done